### PR TITLE
509 the next picture button in the pictureview should trigger a fetch of the next batch of pictures

### LINKED ISSUE #509: the next picture button in the pictureview should trigger a fetch of the next batch of pictures
if you enter an archive for example only a certain number of pictures are loaded, if you then go into the pictureview on a picture and navigate to the last initially loaded picture, no new pictures are loaded. You have to exit the picture view and scroll down to trigger a fetch of the next batch of pictures. This is unintuitive and users might think that there simply are no more pictures left even though they are. This should be fixed

### PR DESCRIPTION
**Pull request by MariusDoe**
Closes #509

The redirect from the `PictureOverview` to the corresponding show-more view takes a short while,
but I think it's okay as that also signals to the user that an automatic redirect happened.

### CODE REVIEW

**Review by olschulz — CHANGES_REQUESTED** [review]
Sometimes when fetching more than the first 10 pictures, there is a reload loop that causes the site to rerender multiple times per second and then crashes the whole tab. I can't reliably reproduce it, but it happened atleast twice now.

image

File: projects/bp-gallery/src/components/common/picture-gallery/PictureScrollGrid.tsx [hunk 1]
```diff
@@ -70,6 +70,29 @@ const PictureScrollGrid = ({
...
+      return;
+    }
+    let fetchCount = NUMBER_OF_PICTURES_LOADED_PER_FETCH;
+    if (maxNumPictures && pictures) {
+      fetchCount = Math.min(maxNumPictures - pictures.length, NUMBER_OF_PICTURES_LOADED_PER_FETCH);
+    }
+    if (fetchCount > 0) {
+      setIsFetching(true);
+      // @ts-ignore
+      // eslint-disable-next-line @typescript-eslint/no-unsafe-call
```
Comment (ole1711) [inline projects/bp-gallery/src/components/common/picture-gallery/PictureScrollGrid.tsx:84]: why are you using ts-ignore and eslint disable here?

Comment (MariusDoe) [reply projects/bp-gallery/src/components/common/picture-gallery/PictureScrollGrid.tsx:84]: I just refactored the code (you'll find it in the old code, too). But you are right, it's not needed and I removed it :)

**Review by ole1711 — APPROVED** [review]

**Review by ole1711 — APPROVED** [review]

**Review by ole1711 — CHANGES_REQUESTED** [review]
bruh nice bug fix

**Comment by MariusDoe** [comment]
Once #559 is merged, the PictureView there should also have this behaviour.

**Review by olschulz — APPROVED** [review]
very cool 🐘👍🏻

**Review by ole1711 — APPROVED** [review]

**Pull request merged**